### PR TITLE
update checkstyle in gradle project

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/java.header [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * <p/>
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/java.header]: I've this question for a long time, should `<p/>` be included in copyright header?

Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/java.header]: We shall remove <p> tag in another PR which will remove all <p> tags in all java headers.

File: PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle-suppressions.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE suppressions PUBLIC
+    "-//Checkstyle//DTD Suppressions 1.1//EN"
+    "https://checkstyle.org/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+  <suppress files="HelpMojo.java" checks="[a-zA-Z0-9]*"/>
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle-suppressions.xml]:
```suggestion
  <suppress files="FileToSuppress.java" checks="[a-zA-Z0-9]*"/>
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -1,114 +1,247 @@
...
+            - import statements
+            - long URLs inside comments
+        -->
 
-        <module name="LeftCurly"/>
-        <module name="NeedBraces"/>
-        <module name="RightCurly"/>
+        <property name="ignorePattern"
+                    value="${com.puppycrawl.tools.checkstyle.checks.sizes.LineLength.ignorePattern}"
+                    default="^(package .*;\s*)|(import .*;\s*)|( *\* *https?://.*)$"/>
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle.xml]: Can we add a regex, such as `\w+://.*` to ignore URI string (like: wasbs://xxx, adfs://xxx, webhdfs://xxx)?

File: PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle.xml [hunk 4]
```diff
@@ -1,114 +1,247 @@
...
+            <!-- Checks for placement of the left curly brace ('{'). -->
+            <property name="severity" value="warning"/>
         </module>
+
         <module name="RightCurly">
-            <property name="id" value="RightCurlySame"/>
-            <property name="tokens"
-                      value="LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_DO"/>
+            <!-- Checks right curlies on CATCH, ELSE, and TRY blocks are on
+            the same line. e.g., the following example is fine:
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle.xml]: Thanks a lot for sample codes!

File: PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle.xml [hunk 5]
```diff
@@ -117,33 +250,47 @@
             <message key="ws.notFollowed"
                      value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as '{}' when not part of a multi-block statement (4.1.3)"/>
             <message key="ws.notPreceded"
-                     value="WhitespaceAround: ''{0}'' is not preceded with whitespace."/>
+                     value="WhitespaceAround: ''{0}'' is not preceded with whitespace."/>
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/config/checkstyle/checkstyle.xml]: Whitespace rules ending with whitespaces. 😄

**Comment by andxu** [comment]
Close it since there are too many violations now, we should add rules one by one

**Pull request closed without merging**